packages/registry/README.md

```diff
@@ -0,0 +1,190 @@
+
+The add-ons are registered in the order they are found in the `addons` key. The last add-on takes precedence over the others. This means that if you configure something in `acme-volto-foo-addon` and then the same in `collective-another-volto-addon`, the latter configured will win and that configuration applied.
+
+The default export of any add-on main `index.js` file should be a function with the signature `config => config`. That is, it should take the configuration registry object and return it, possibly mutated or changed.
+
+## Configuration registry
+
+The add-on registry is complemented with the configuration registry.
+
+It is a facility that stores app configuration to be shared in the app. This is where the extensibility and pluggability story for our app lies.
+
+Let's say that your app is the UI of a content management system (CMS). This CMS uses blocks as its main fundamental unit of content. The pages that the CMS builds are made up of these blocks. The CMS has some basic available blocks, but it's a requirement that integrators are able to register more blocks in a pluggable way.
+
+This app will use the add-on registry to extend the basic CMS capabilities, so an external add-on will be able to sum up with their own add-on(s) to the basic CMS ones.
+
+Let's assume we've defined a key in the registry `config.blocks.blocksConfig` and defined a way to register the available blocks in the CMS as the keys in that object in the configuration registry:
+
+```js
+  config.blocks.blocksConfig.faq_viewer = {
+    id: 'faq_viewer',
+    title: 'FAQ Viewer',
+    edit: FAQBlockEdit,
+    view: FAQBlockView,
+    icon: chartIcon,
+    group: 'common',
+    restricted: false,
+    mostUsed: true,
+    sidebarTab: 1,
+  };
+  ```
+
+The configuration registry will have other keys already set by default which will compose the initial set of basic blocks used by the CMS. The CMS will populate the available blocks in the CMSUI properly to be used by the users.
+
+The add-on is meant to extend the initial configuration, so from the default export function of our add-on we should provide the configuration of the new block:
+
+```ts
+export default function applyConfig(config: ConfigData) {
+  config.blocks.blocksConfig.faq_viewer = {
+    id: 'faq_viewer',
+    title: 'FAQ Viewer',
+    edit: FAQBlockEdit,
+    view: FAQBlockView,
+    icon: chartIcon,
+    group: 'common',
+    restricted: false,
+    mostUsed: true,
+    sidebarTab: 1,
+  };
+
+  return config;
+}
+```
+
+Once the app starts, the add-on registry will execute in order all the registered add-ons default export functions, configuring the new block and make it available to the CMS when it asks to the configuration registry for it.
```

Comment:
Too many pronouns make it difficult to understand what "it" refers to. Here is my guess.

```suggestion
Once the app starts, the add-on registry will execute, in order, all the registered add-ons' default export functions, configuring the new block, and the add-on available to the CMS when it asks the configuration registry for it.
```